src/main/kotlin/org/rust/ide/typing/paste/RsConvertJsonToStructCopyPasteProcessor.kt

```diff
@@ -221,13 +260,31 @@ fun generateStruct(
 
 private val NON_IDENTIFIER_REGEX: Regex = Regex("[^a-zA-Z_0-9]")
 
-fun normalizeFieldName(field: String): String {
+private fun normalizeFieldName(field: String): String {
     var name = field.replace(NON_IDENTIFIER_REGEX, "_")
     if (name.getOrNull(0)?.isDigit() == true) {
         name = "_$name"
     }
 
-    return name.toSnakeCase(false).escapeIdentifierIfNeeded()
+    name = name.toSnakeCase(false)
+    if (name.all { it == '_' }) {
+        name += "field"
+    }
+
+    return name.escapeIdentifierIfNeeded()
+}
+
+private fun createFieldName(field: String, generatedFieldNames: Set<String>): String {
+    val normalizedName = normalizeFieldName(field)
+    if (normalizedName !in generatedFieldNames) return normalizedName
+
+    var index = 0
+    var name = normalizedName
+    while (name in generatedFieldNames) {
+        name = "${normalizedName}_$index"
+        index += 1
+    }
+    return name
```

Comment:
```suggestion
    return generateSequence(0) { it + 1 }
        .map { "${normalizedName}_$it" }
        .first { it !in generatedFieldNames }
```

minor: more functional approach